libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1458,18 +1523,25 @@ string YulUtilFunctions::readFromCalldata(Type const& _type)
 	return readFromMemoryOrCalldata(_type, true);
 }
 
-string YulUtilFunctions::updateStorageValueFunction(Type const& _type, std::optional<unsigned> const& _offset)
+string YulUtilFunctions::updateStorageValueFunction(
+	Type const& _fromType,
+	Type const& _toType,
+	std::optional<unsigned> const& _offset
+)
 {
 	string const functionName =
 		"update_storage_value_" +
 		(_offset.has_value() ? ("offset_" + to_string(*_offset)) : "") +
-		_type.identifier();
+		_fromType.identifier() +
+		"_to_" +
+		_toType.identifier();
 
 	return m_functionCollector.createFunction(functionName, [&] {
-		if (_type.isValueType())
+		if (_toType.isValueType())
 		{
-			solAssert(_type.storageBytes() <= 32, "Invalid storage bytes size.");
-			solAssert(_type.storageBytes() > 0, "Invalid storage bytes size.");
+			solAssert(_fromType.isImplicitlyConvertibleTo(_toType), "");
```

Comment:
So this will actually be called for distinct, but convertible types? Then maybe we should spend a second thinking about whether there should be some conversion functions involved in the process here :-)...

Reply:
Do you have an example case for which this is called with ``_fromType.isImplicitlyConvertibleTo(_toType)``, but with ``_fromType != _toType``?

Reply:
There was few tests with integer and rational literals failed test for types being equal. So, this can be called for implicitly convertible value types...

Reply:
For example: `uint x = 42;`

Reply:
Ok, that makes sense. Ok, I can't come up with a case that would be problematic here, and I guess the ``<prepare>`` and ``<update>`` below actually do some kind of conversion anyways, so this should be fine.